activestorage/lib/active_storage/validations.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+require "active_model"
+require "active_support/concern"
+require "active_support/core_ext/array/wrap"
+require "active_storage/validations/base_validator"
+require "active_storage/validations/attachment_size_validator"
+require "active_storage/validations/attachment_content_type_validator"
+
+module ActiveStorage
+  # Provides the class-level DSL for declaring ActiveStorage validations
+  module Validations
+    extend ActiveSupport::Concern
+
+    included do
+      extend  HelperMethods
+      include HelperMethods
+    end
+
+    module ClassMethods
+      # A helper method to run various Active Storage attachment validators.
+      #
+      # Effectively the same as the <tt>ActiveModel::Validations#validates</tt>
+      # method but more readable since it does not require the +attachment_+
+      # prefix for its keys.
+      #
+      #   validates_attachment :avatar, size: { in: 2..4.megabytes }
+      #   validates_attachment :avatar, content_type: { in: "image/jpeg" }
+      #
+      # Like the <tt>ActiveModel::Validations#validates</tt>, it also
+      # supports shortcut options which can handle ranges, arrays, and strings.
+      #
+      #   validates_attachment :avatar, size: 2..4.megabytes
+      #   validates_attachment :avatar, content_type: "image/jpeg"
+      #
+      # When using shortcut form, ranges and arrays are passed to the
+      # validator as if they were specified with the +:in+ option, while other
+      # types including regular expressions and strings are passed as if they
+      # were specified with +:with+.
```

Comment:
This is a bit better to read IMO, since it doesn't repeat the same word twice:
```suggestion
      # were specified using +:with+.
```